File: mvvmwithoutcachingjavadagger/src/main/java/com/mfcwl/dagger/di/ActivityBuildersModule.java
package com.mfcwl.dagger.di;

import com.mfcwl.dagger.di.auth.AuthModule;
import com.mfcwl.dagger.di.auth.AuthScope;
import com.mfcwl.dagger.di.auth.AuthViewModelsModule;
import com.mfcwl.dagger.di.main.MainFragmentBuildersModule;
import com.mfcwl.dagger.di.main.MainModule;
import com.mfcwl.dagger.di.main.MainScope;
import com.mfcwl.dagger.di.main.MainViewModelsModule;
import com.mfcwl.dagger.ui.auth.AuthActivity;
import com.mfcwl.dagger.ui.auth.AuthViewModel;
import com.mfcwl.dagger.ui.main.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuildersModule {

    @AuthScope
    @ContributesAndroidInjector(
            modules = {AuthViewModelsModule.class, AuthModule.class}
    )
    abstract AuthActivity contributeAuthActivity();

    @MainScope
    @ContributesAndroidInjector(
            modules = {MainFragmentBuildersModule.class, MainViewModelsModule.class, MainModule.class}
    )
    abstract MainActivity contributeMainActivity();
}

File: powerfulandroidapps_processdeathissuefix/src/main/java/com/mfcwl/library/powerfulandroidapps_processdeathissuefix/di/ActivityBuildersModule.kt
package com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di

import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.auth.AuthFragmentBuildersModule
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.auth.AuthModule
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.auth.AuthScope
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.auth.AuthViewModelModule
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.main.MainFragmentBuildersModule
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.main.MainModule
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.main.MainScope
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di.main.MainViewModelModule
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.ui.auth.AuthActivity
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.ui.main.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuildersModule {

    @AuthScope
    @ContributesAndroidInjector(
        modules = [AuthModule::class, AuthFragmentBuildersModule::class, AuthViewModelModule::class]
    )
    abstract fun contributeAuthActivity(): AuthActivity

    @MainScope
    @ContributesAndroidInjector(
        modules = [MainModule::class, MainFragmentBuildersModule::class, MainViewModelModule::class]
    )
    abstract fun contributeMainActivity(): MainActivity

}